### PR TITLE
splitRaster default to memory

### PR DESCRIPTION
**Pull request by ianmseddy**
changed behavior of splitRaster path argument so does not write to disk if path = NA

### CODE REVIEW

**Comment by eliotmcintire** [comment]
from Travis: You need to give a default value to path, if you are going to use it in a function.
```
  Loading required package: SpaDES.tools
  ── 1. Error: splitRaster and mergeRaster work on small in-memory rasters (@test-
  argument "path" is missing, with no default
  1: splitRaster(r, nx, ny, c(3L, 4L)) at testthat/test-splitRaster.R:84
  2: splitRaster(r, nx, ny, c(3L, 4L))
```

**Comment by achubaty** [comment]
Based on your changes, the default path value should be `NA`. Alternatively you could use `NULL` and check for for `is.null` instead of `is.na`.

**Comment by ianmseddy** [comment]
Okay, I think I fixed that. Some of the tests were failing too because writeRaster now apparently accepts "INT" and "FLT" as raster datatypes.

File: tests/testthat/test-splitRaster.R [hunk 1]
```diff
@@ -22,7 +22,7 @@ test_that("splitRaster and mergeRaster work on small in-memory rasters", {
   extent(r) <- extent(xmin(r) - 30, xmax(r) - 30, ymin(r) - 20, ymax(r) - 20)
 
   # no buffer
-  y0 <- splitRaster(r, nx, ny)
+  y0 <- splitRaster(r, nx, ny, path = file.path(getwd(), "red"))
```
Comment (achubaty) [inline tests/testthat/test-splitRaster.R]: make sure this is a temp location

Comment (ianmseddy) [reply tests/testthat/test-splitRaster.R]: The working directory is set to the temp directory on line 7, just before this. Do you want me to call the temp drive directly in the path argument of line 25?

Comment (achubaty) [reply tests/testthat/test-splitRaster.R]: yes, easier to keep track of what's going on later if the tempdir used directly

**Pull request merged**